abci.go

```diff
@@ -78,13 +78,13 @@ func (c CheckResult) ToABCI() abci.ResponseCheckTx {
 		Data:      c.Data,
 		Log:       c.Log,
 		GasWanted: c.GasAllocated,
-		Fee:       common.KI64Pair{Value: c.GasPayment},
+		GasUsed:   c.GasPayment,
```

Comment:
I think this is not the proper field....

If I understand cosmos/tendermint design properly:
`GasWanted` is returned from `CheckTx`
`GasUsed` should be returned from `DeliverTx` (no idea why... we can just return same as GasWanted.

The confusion here: https://github.com/tendermint/tendermint/blob/master/abci/types/types.proto#L165-L183 is due to Jae insisting the two return values are the same, even if handled differently.

Reply:
The only real reference is what cosmos-sdk does:

https://github.com/cosmos/cosmos-sdk/blob/develop/types/result.go

https://github.com/cosmos/cosmos-sdk/blob/d666658ca9ed0ef5664a58002246f4a4915c08ac/baseapp/baseapp.go#L563-L570

Let's just follow them

Reply:
The 2 different attributes make sense make sense as an optimisation where the concrete value is computational expensive to be calculated or maybe not fully definable unless processed. Now we can have `GasWanted` >=`GasUsed`.